src/main/resources/templates/gardens/list.html

```diff
@@ -12,8 +12,50 @@
 <body>
 <div th:replace="fragments :: page-header"></div>
 <div class="container body-content">
+    <div>
+        <!-- Search form -->
+        <form th:action="@{/user/gardens/search}" method="GET">
+            <div style="margin-top: 5px">
+            <input class="form-control mr-sm-2"
+                   th:name="searchTerm"
+                   type="text"
+                   th:value="${searchTerm}"
+                   aria-label="Search">
+            </div>
+            <div class="form-group" style="margin-bottom: 0;">
+                <label th:for="features">Features: </label>
+                <th:block th:each="feature : ${features}">
+                    <input type="checkbox"
+                           th:name="featureIds"
+                           th:value="${feature.id}"
+                           th:if="${selectedFeatures.contains(feature)}"
+                           checked>
+                    <input type="checkbox"
+                           name="featureIds"
+                           th:value="${feature.id}"
+                           th:unless="${selectedFeatures.contains(feature)}">
+                    <label th:text="${feature.name} + '  '"></label>
+                </th:block>
+                <p>Note: all gardens are searched if no categories are selected.</p>
+            </div>
+            <div>
```

Comment:
are the empty div tags here and closing it on 43 needed for anything? If they are needed, what are they doing?

Reply:
I thought I was using them around the button but I removed them and nothing changed so yeah they were definitely unneeded.